docs/essentials/parameterized-promotions.md

```diff
@@ -0,0 +1,193 @@
+---
+Description: Use parameterized promotions to improve code reusability and to dynamically configure your promotions and deployments.
+---
+
+# Parameterized Promotions
+
+The **parameterized promotions** feature allows users to pass parameters to promotion
+pipelines. You can do this either via a form in the UI, 
+via the Semaphore API, or on auto-promotions by setting up default values.
+
+The core advantage of this feature is **code reusability**. It is no longer necessary to create many promotion pipelines with almost identical configurations. Users can now create a single pipeline and rely on parameters for smaller but critical differences.
+
+## Configuring the promotion parameters
+Let's start with defining the parameters that our promotion will support. 
+
+In your `.yml` file, when defining the promotion you can set up
+as many parameters as you want. 
+
+Possible options include:
+
+- **required**: values (true/false) - If the parameter is required and
+default_value is not set the promotion won’t start.
+- **default_value** - Can only be set if the parameter is required, it
+is a default value of the parameter.
+- **options** - Use this to define a list of all possible values and the user will have to select one of them from the dropdown menu. If omitted, the user will be able to input any value via the input box.
+- **description** - This is a description text that gets rendered on
+the promotion UI (when a user selects values)
+- **name** - This is the name of the parameter and the exported
+environment variable (referenced later in the promoted
+pipelines).
+
+Here is an example of setting the two parameters for a single promotion:
+``` yaml
+# .semaphore/semaphore.yml
+version: v1.0
+name: Parameterized Promotion
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu2004
+blocks:
+  - name: Promotions
+    task:
+      jobs:
+        - name: Everything
+          commands:
+            - echo 'Running tests'
+promotions:
+  - name: Deploy
+    pipeline_file: deploy.yml
+    # Define promotion parameters
+    parameters:
+      env_vars:
+        - required: true
+          options:
+            - Stage
+            - Production
+          default_value: Stage
+          description: Where to deploy?
+          name: ENVIRONMENT
+       
+        - required: false
+          description: Release name?
+          name: RELEASE
+```
+
+You can also use the in-app Workflow Editor to define the parameters for the promotion like this:
+
+<img style="box-shadow: 0px 0px 5px #ccc" src="/essentials/img/parameterized-promotions/wf-editor.png" alt="Parameterized promotions in Workflow Editor">
+
+
+## Using the parameter values in promoted pipeline
+Now that we have defined our parameters, it is time to define how they are used in the promoted pipeline. 
+
+Promotion parameters can be used in the following way: 
```

Comment:
You can use promotion parameters in the following way:

passive to active voice.